wandb/sdk/wandb_init.py

```diff
@@ -522,17 +525,60 @@ def _log_setup(self, settings: Settings) -> None:
         logger.info(f"Logging internal logs to {settings.log_internal}")
 
     def _make_run_disabled(self) -> Run:
+        """Create a disabled run object.
+
+        This enables the user to call wandb.init(mode="disabled") and still have a Run object.
+        We ensure that the attributes and methods of the run object are available, but do nothing.
```

Comment:
Should probably say more here. How does this help users?

Maybe something like

> Although it's inadvisable, this allows users to pass a no-op Run object to APIs that expect and call methods on a Run. Without this, users encounter attribute errors.

Even better if we have any specific examples.